### ISSUE TITLE
sap.m.Menu itemSelected event parameter item is typed too broadly

### ISSUE THREAD

**Issue by edemuck** [UI5/typescript#629]
### Problem
In the generated typings for `sap.m.Menu`, the event parameter `item` of `Menu$ItemSelectedEventParameters` is typed as `IMenuItem`.

However, for the `itemSelected` event, the selected item should be a concrete `sap.m.MenuItem`.

### Current typing
In `sap.m.d.ts`, the type is defined as:

```ts
export interface Menu$ItemSelectedEventParameters {
  item?: IMenuItem;
}
```

### Expected typing
The item property should be typed as:
```ts
item?: MenuItem;
```

or equivalently sap.m.MenuItem, if that is the preferred style in the generated declarations.

### Why this matters
IMenuItem is too generic for consumers of the itemSelected event and makes the event payload less precise than the runtime contract. This can lead to unnecessary type checks or missing access to MenuItem-specific APIs in application code.

### Suggested fix
Update the generated typings for Menu$ItemSelectedEventParameters so that item uses MenuItem instead of IMenuItem.

**Comment by codeworrior** [comment]
This complaint better goes to openui5 as the source of type information is there. It's not related to the mapping done by the dts-generator.

Also see our hint regarding this in https://github.com/UI5/typescript#support.

**Comment by edemuck** [comment]
Thanks for taking it up as an OpenUI5 bug!
